store.go

```diff
@@ -3713,11 +3724,10 @@ func copyDigestSlice(slice []digest.Digest) []digest.Digest {
 // copyStringInterfaceMap still forces us to assume that the interface{} is
 // a non-pointer scalar value
 func copyStringInterfaceMap(m map[string]interface{}) map[string]interface{} {
-	ret := make(map[string]interface{}, len(m))
-	for k, v := range m {
-		ret[k] = v
+	if m == nil {
+		return make(map[string]interface{}, len(m))
 	}
-	return ret
+	return maps.Clone(m)
```

Comment:
(Long-term, I think it will pay off to split bug fixes from no-effect refactors into separate commits.)

Reply:
Does the `copyStringInterfaceMap` function have a specific purpose, or should it also be replaced by a generic function as you suggest in the previous comment?

Reply:
I don’t know that there is anything special here — this is all code written before Go had generics.